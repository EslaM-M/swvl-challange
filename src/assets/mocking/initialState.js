// import route from "../data/route";
import tripInformation from "../data/trip-information";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
