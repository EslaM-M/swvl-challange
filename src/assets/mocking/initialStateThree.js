// import route from "../data/route";
import tripInformation from "../data/trip-information";
import fullBooking from "../data/full-booking-list";
export const initialStateThree = {
  bookings: fullBooking,
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
