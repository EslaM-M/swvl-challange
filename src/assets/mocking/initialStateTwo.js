// import route from "../data/route";
import tripInformation from "../data/trip-information";
import bookings from "../data/booking";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route,
  bookings: bookings,
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
