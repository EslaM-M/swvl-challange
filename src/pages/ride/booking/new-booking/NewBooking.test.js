import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBooking from "./index";
import  {initialState}  from "../../../../assets/mocking/initialState";
describe("NewBooking Component", () => {
  let realUseContext;
  let useContextMock;
  // Setup mock
  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
    useContextMock.mockReturnValue({
      state: initialState
    });
  });
  // Cleanup mock
  afterEach(() => {
    React.useContext = realUseContext;
  });

  test("Component snapshot render correctly", () => {
    const container = render(<NewBooking show={true} />);
    expect(container).toMatchSnapshot();
  });

  test("Component save changes disabled if no user input", async () => {
    const { getByText } = render(<NewBooking show={true} />);
    const action = getByText("Save Changes");
    expect(action.disabled).toBe(true);
  });
  test("Component select station have the exact number of options", async () => {
    const { baseElement } = render(<NewBooking show={true} />);
    const options = baseElement.querySelectorAll("option");
    expect(options.length).toBe(11);
  });

  test("Component Save Changes Enabled after add both (station and payment)", async () => {
    const { getByText, getByTestId } = render(<NewBooking show={true} />);
    fireEvent.change(getByTestId("select-station"), { target: { value: 1 } });
    fireEvent.click(getByTestId("test-credit-value"));
    const action = getByText("Save Changes");
    expect(action.disabled).toBe(false);
  });

  test("Component Save Changes Disabled after station ", async () => {
    const { getByText, getByTestId, container } = render(
      <NewBooking show={true} />
    );
    fireEvent.change(getByTestId("select-station"), { target: { value: 1 } });
    const action = getByText("Save Changes");
    expect(action.disabled).toBe(true);
  });

  test("Component Save Changes Disabled after payment type ", async () => {
    const { getByText, getByTestId } = render(<NewBooking show={true} />);
    fireEvent.click(getByTestId("test-credit-value"));
    const action = getByText("Save Changes");
    expect(action.disabled).toBe(true);
  });
});
