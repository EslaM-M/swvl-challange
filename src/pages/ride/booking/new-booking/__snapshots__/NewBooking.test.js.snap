// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBooking Component Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-hidden="true"
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              New Booking
            </div>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
              <span
                class="sr-only"
              >
                Close
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="newBookingForm.pickupStation"
                >
                  Example select
                </label>
                <select
                  class="form-control"
                  data-testid="select-station"
                  id="newBookingForm.pickupStation"
                  name="pickupStation"
                >
                  <option
                    value=""
                  >
                    Select Station
                  </option>
                  <option
                    value="1"
                  >
                    station1
                  </option>
                  <option
                    value="2"
                  >
                    station2
                  </option>
                  <option
                    value="3"
                  >
                    station3
                  </option>
                  <option
                    value="4"
                  >
                    station4
                  </option>
                  <option
                    value="5"
                  >
                    station5
                  </option>
                  <option
                    value="6"
                  >
                    station6
                  </option>
                  <option
                    value="7"
                  >
                    station7
                  </option>
                  <option
                    value="8"
                  >
                    station8
                  </option>
                  <option
                    value="9"
                  >
                    station9
                  </option>
                  <option
                    value="10"
                  >
                    station10
                  </option>
                </select>
              </div>
              <div
                class="form-group row"
              >
                <legend
                  class="form-label col-form-label col-sm-4"
                >
                  Payment Type
                </legend>
                <div
                  class="col-sm-8"
                >
                  <div
                    class="form-check"
                  >
                    <input
                      class="form-check-input"
                      data-testid="test-cash-value"
                      name="paymentType"
                      type="radio"
                      value="cash"
                    />
                    <label
                      class="form-check-label"
                      title=""
                      type="checkbox"
                    >
                      Cash
                    </label>
                  </div>
                  <div
                    class="form-check"
                  >
                    <input
                      class="form-check-input"
                      data-testid="test-credit-value"
                      name="paymentType"
                      type="radio"
                      value="credit"
                    />
                    <label
                      class="form-check-label"
                      title=""
                      type="checkbox"
                    >
                      Credit
                    </label>
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              disabled=""
              type="button"
            >
              Save Changes
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
